src/index.ts

```diff
@@ -207,13 +211,33 @@ function startApp(appSchema, path: string) {
     }
   })
 
+  const graphqlUpload = graphqlUploadExpress({
+    maxFileSize: GRAPHQL_UPLOAD_MAX_FILE_SIZE_IN_BYTES,
+    maxFiles: GRAPHQL_UPLOAD_MAX_FILES,
+  })
+
   app.use("/batch", bodyParser.json(), graphqlBatchHTTPWrapper(graphqlServer))
-  app.use(graphqlServer)
+  app.use(graphqlUpload, graphqlServer)
 
   if (enableSentry) {
     app.use(Sentry.Handlers.errorHandler())
   }
 
+  // Global error handler
+  app.use((error, _req, res, next) => {
+    if (res.headersSent) {
+      return next(error)
+    }
+
+    const status = error?.status ?? 500
+    const message = error?.message ?? "Something went wrong"
+
+    return res.status(status).json({
+      message,
+      stack: !PRODUCTION_ENV ? error.stack.split("\n") : null,
+    })
+  })
+
```

Comment:
**question**: Should global error handler be used in Express to return error message and stack trace (see screenshot #1, #2)? Otherwise, we will always get 404 response code and the corresponding message text for this error (see attached screenshot #3).

On the one hand, it can help developers to quickly understand what is working wrong and find error. For example, as it was in my cases when I started working with `graphql-upload` and always got `404 Cannot POST /v2` in response to requests where I sent the file. It turned out that I used the wrong query structure and an error was thrown in the `graphql-upload` middleware for this reason. But due to the fact that we did not have a global error handler, the default `404 error` was always returned.

But on the other hand, won't I break something in another place or in another application if I add these changes? What do you think about this? Or maybe add a global error handler only for staging/development environments?

### Screenshot 1
<img width="820" alt="error-1" src="https://user-images.githubusercontent.com/3513494/164517251-c2ff0b31-c794-49d3-bc06-d29f5416b8a6.png">

### Screenshot 2
<img width="820" alt="error-2" src="https://user-images.githubusercontent.com/3513494/164517274-8b9f7f47-c116-4275-be2d-223cc4659425.png">


### Screenshot 3
<img width="820" alt="error-4" src="https://user-images.githubusercontent.com/3513494/164517289-25e32ce1-c1d7-4451-a1af-2aa77a88f111.png">